src/common/transformations/src/transformations/op_conversions/convert_fc_to_compressed.cpp

```diff
@@ -0,0 +1,185 @@
+
+            auto new_shape = (has_transpose || !grouped)
+                                 ? ov::Shape{current_shape[0] * current_shape[1], current_shape[2]}
+                                 : ov::Shape{current_shape[0], current_shape[1] * current_shape[2]};
+
+            return std::make_shared<ov::op::v0::Constant>(*constant, new_shape);
+        };
+
+        auto convert_u4const_to_u8 = [convert_u4zp_to_u8](std::shared_ptr<ov::Node> node) {
+            auto constant = std::dynamic_pointer_cast<ov::op::v0::Constant>(node);
+            if (constant->get_element_type() != ov::element::u4 || !convert_u4zp_to_u8)
+                return std::dynamic_pointer_cast<ov::Node>(constant);
+            return std::dynamic_pointer_cast<ov::Node>(std::make_shared<ov::op::v0::Convert>(node, ov::element::u8));
+        };
+
+        const ov::Output<Node>& fc_input_a = fc->input(0).get_source_output();
+        const auto& scale = reshape_const_to_2d(pattern_map.at(mul_const_m).get_node_shared_ptr());
+        std::shared_ptr<ov::Node> optional_zero_point = nullptr;
+
+        const bool with_zero_point =
+            pattern_map.count(sub_no_convert_m) > 0 || pattern_map.count(sub_with_convert_m) > 0;
+        if (with_zero_point) {
+            // WA: Convert ZP to u8 for OneDNN case to avoid u4 reorder
+            optional_zero_point =
+                convert_u4const_to_u8(reshape_const_to_2d(pattern_map.at(sub_const_m).get_node_shared_ptr()));
+        }
+
+        std::shared_ptr<ov::Node> fc_input_b = reshape_const_to_2d(pattern_map.at(weights_m).get_node_shared_ptr());
+        std::shared_ptr<ov::Node> fc_input_scale = scale;
+        std::shared_ptr<ov::Node> fc_input_zp = optional_zero_point;
+        std::shared_ptr<ov::Node> fc_input_bias = pattern_map.at(bias_m).get_node_shared_ptr();
+        std::vector<std::shared_ptr<ov::Node>> result_nodes = {};
+        if (has_transpose) {
+            const auto& transpose = pattern_map.at(transpose_m).get_node_shared_ptr();
+            std::shared_ptr<ov::Node> transpose_const = pattern_map.at(transpose_const_m).get_node_shared_ptr();
+            if (ov::shape_size(transpose_const->get_shape()) != fc_input_b->get_output_partial_shape(0).size()) {
+                std::vector<int32_t> new_order(fc_input_b->get_output_partial_shape(0).size());
+                std::iota(new_order.begin(), new_order.end(), 0);
+                std::swap(new_order[new_order.size() - 1], new_order[new_order.size() - 2]);
+                transpose_const =
+                    std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{new_order.size()}, new_order);
+            }
+
+            fc_input_b = transpose->clone_with_new_inputs({fc_input_b->output(0), transpose_const});
+            ov::disable_constant_folding(fc_input_b);
+            result_nodes.push_back(fc_input_b);
+            fc_input_scale = transpose->clone_with_new_inputs({scale->output(0), transpose_const});
```

Comment:
In case of scalar scale we don't need to transpose it. Should we add the corresponding check as it is done for zero point?

Reply:
see above